### ISSUE TITLE
News users automatically set as wp subscribers

### ISSUE THREAD

**Issue by kevindherman** [proudcity/wp-proudcity#1936]
**Source**
Holyoke and I've gotten a few tickets from San Rafael about this but it doesn't seem as consistent. 

**Describe the bug**
When an admin creates a new user through my.proudcity.com dashboard, including the info@proudcity.com account, the new users is always a subscriber when you look at the WP permissions. Customers don't have access to the wp user interface to change subscribers to editors so we have to do it after they have submitted a ticket. 

**To Reproduce**
1. Go to my.proudcity.com
2. Go to users
3. Invite new user
4. You'll then see the user (WP side) is listed as a subscriber not editor https://www.holyoke.org/wp-admin/users.php?role=subscriber
5. Edit the user to editor to get them the right permissions (but this shouldn't happen).

**Expected behavior**
Have new users set to wp editor permissions. 

**Additional context**
This seems isolated to Holyoke as being pervasive to even the info@proudcity.com account.

**Comment by lukefretwell** [comment]
@kevindherman @jlyon this happened with SR today.

**Comment by kevindherman** [comment]
@lukefretwell SR and Holyoke are the main culprits.

**Comment by jlyon** [comment]
@curtismchale when the form is submitted in the dashboard, it submits to a node.js api: https://github.com/proudcity/proudcity-api-swagger/blob/master/api/controllers/user.js#L538, which then runs `wp user create` on the relevant site (https://github.com/proudcity/proudcity-api-swagger/blob/master/api/controllers/user.js#L624). The logs for these wp command runs can be found on the dashboard under Settings > Queue. 

In holyoke's case,  it looks like the command is running with the proper --role flag. https://my.proudcity.com/sites/5e6b0b0a7982dd0170307683/queue:
```
cmd prod holyokema wp user create ***@holyoke.org ***@holyoke.org --send-email=0 --porcelain --role=editor
```
which corresponds to 
```
wp user create --allow-root holyokema wp user create ***@holyoke.org ***@holyoke.org --send-email=0 --porcelain --role=editor
```

I would probably start by debugging around those wp-cli calls to see if we can get to the bottom of it that way. lmk if I can help.

**Comment by kevindherman** [comment]
@curtismchale I don't know about SR today but looking through support tickets I know that

[biermanns@holyoke.org](mailto:biermanns@holyoke.org) had this problem when created by [anderson-burgosj@holyoke.org](mailto:anderson-burgosj@holyoke.org) on July 20th.

Does that help?

**Comment by curtismchale** [comment]
@kevindherman It's a start but if it's some email-specific thing then I'll need a lot more examples to figure out what about the emails is the issue.

I'm trying to figure out what about these two sites (or one site) is different that causes it not to work. It seems to work just fine on many other sites.

**Comment by curtismchale** [comment]
See this list provided by Kevin in [another ticket by accident](https://github.com/proudcity/wp-proudcity/issues/1934#issuecomment-1144879476).

- [Kate.Anderson@cityofsanrafael.org](mailto:Kate.Anderson@cityofsanrafael.org) on San Rafael
- rafaelh@co.williams.nd.us on Williams County
- ryanm@co.williams.nd.us on Williams County
- ecochran@marincounty.org on Marin Recovers

**Comment by curtismchale** [comment]
- [ ] capture command output see: https://make.wordpress.org/cli/handbook/references/shell-friends/

Blocked by https://github.com/proudcity/wp-proudcity/issues/2033

**Comment by kevindherman** [comment]
@curtismchale @lukefretwell I was able to figure out one way in which this is happening.

1. New user is create but the admin does NOT set a role before inviting
2. WP site setting default to new users being WP subscribers 

All of PC roles are based off of variations of WP editor. So if no PC role defaults to subscriber, when the admin updates the permissions on PC the changes aren't reflected because the WP role is stuck as a subscriber.

Changing the default new user to editor rather than subscriber would likely resolve this but seems not smart from a security standpoint. Maybe there's another way by not allowing new users to be created without selecting a CMS role.

**Comment by curtismchale** [comment]
@kevindherman We could check to make sure that a value is set when a user is created and **not** allow them to create user without a value set. That would require a change to the template and some wording to indicate what the issue is.

We could also set the lowest permission user role in our end as the default if nothing is selected. This would mean they have minimum access to the site and in theory their access could be changed from our Admin dashboard.

Either way works. I wonder if it would be good to add a link/popup/something by the Content Roll and Payment Role dropdown to explain what each role lets a user do so that customers know what they're allowing people to access.

**Notes for me**

The form is set in [`app/views/site/users.html`](https://github.com/proudcity/proudcity-dashboard/blob/f92db3b7c03ba6217f48357cc57f57482b2ff610/app/views/site/users.html#L21) and we want to require the Content Role to have a selection made.

Default roles defined in [`app/js/routes/siteRoutes.js`](https://github.com/proudcity/proudcity-dashboard/blob/f92db3b7c03ba6217f48357cc57f57482b2ff610/app/js/routes/siteRoutes.js#L833)

I'm not sure where we connect to WP to send the `wp user create` command as a search for anything like that returns hundreds of results from old deploy notes and files.
